subsys/bluetooth/host/conn.c

```diff
@@ -768,6 +773,12 @@ void bt_conn_process_tx(struct bt_conn *conn)
 	buf = net_buf_get(&conn->tx_queue, K_NO_WAIT);
 	BT_ASSERT(buf);
 	if (!send_buf(conn, buf)) {
+		/* buf is going to be destroyed by the next unref */
+		if (buf->ref == 1 && tx_data(buf)->tx) {
```

Comment:
I'm not sure about the `buf->ref == 1` requirement. I think the TX callbacks must be invoked by whatever process removed `buf` from `tx_queue`. In this code branch, `buf` is not put back on `tx_queue`. Am I wrong?

Reply:
The TX thread does not care if `send_buf()` succeeds or not and does not put the buffer back in the `tx_queue` if it fails.

Reply:
yeah its not put back, I'll remove it.